### DISCUSSION TITLE
scaffold.pl section titles

### DISCUSSION THREAD

**Discussion by Alex-Jordan** [openwebwork/pg#578] [General]
My understanding is that in a scaffold problem, you can provide a title for each section. And it may be common for the title to include some numbering. The first section title might be "1.  Find the coefficients." and the second might be "2. Use the Quadratic Formula". Or something like that.

For upcoming PreTeXt work, I would like to give scaffold section actual numbering, separate from the title. Where we currently have an `h3` for each section title, I would (perhaps naively) change it to an HTML ordered list, and render any title in some stylized way inline with the part number. (Side note: I'm not sure the `h3` is right in the first place for a few reasons.)

Does this raise any flags? One is that problems that already have titles like "Part 1.  Find the coefficients." would now be rendered like "1.  Part 1.  Find the coefficients."  Personally I think that is a small thing and separating the numbering from the title would be worth it. But do people feel otherwise? Are there larger reasons not to automatically number the parts in a scaffolded problem?

**Comment by drgrice1** [comment]
What `h3` are you talking about?  I don't see such a thing in the scaffold.pl macro.

**Comment by drgrice1** [comment]
I see.  You are talking about the macro as it was in WeBWorK 2.15 and before.  In WeBWorK 2.16 it is no longer an `<h3>`.  It is an `<a>`.

**Reply by Alex-Jordan** [reply]
Did it have an `h3` in the early version of 2.16? I believe that I was just looking on my production server, which is early 2.16.

**Comment by drgrice1** [comment]
The scaffold macro was not changed from ww 2.15 until recently.  The pull request that changed it was just merged one week ago.

**Comment by drgrice1** [comment]
Note that you will not be able to change to an HTML ordered list in scaffolds.  That would not work with either the jquery-ui accordion that was used before, or with the bootstrap accordion that is now used.

**Reply by Alex-Jordan** [reply]
OK, thanks that's good to know before I would attempt something. An HTML `ol` is not the important part for what I am thinking about. Just having some sort of automated numbering that is separate from the title is what matters. The simplest would be that it could be automated when the page is built and just hard written into the page. Even if it just pre-concatenated the current title with strings like "1. "

In other words, the goal would be having problem authors just declare "Find the coefficients." as the title instead of "1. Find the coefficients." But the end user still sees numbering.

**Reply by dpvc** [reply]
I think the key line is

https://github.com/openwebwork/pg/blob/a284c3b31a26fb316c2fa6b34246df8ba2895a3e/macros/scaffold.pl#L506

which could be made something like

``` perl
$self->{name} = ($self->{name} ? "$self->{number}. $self->{name}" : "Part $self->{number}:");
```

(untested)

**Comment by drgrice1** [comment]
I don't like the idea of forcing numbering on the scaffold section title.  I prefer having the freedom to put numbering on if I want it, and not to if I don't want numbering.

**Reply by Alex-Jordan** [reply]
How about an option like:

```
Scaffold::Begin(
      numbered => 1,
    );
```

**Comment by drgrice1** [comment]
I am fine with it if it is optional.

**Comment by Alex-Jordan** [comment]
See https://github.com/openwebwork/pg/pull/579.
